### PR TITLE
[web] Implement defaultTargetPlatform MacOS/iOS for web.

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

Returns correct platform when application is running on iOS and MacOS.

## Related Issues

https://github.com/flutter/flutter/issues/43633

## Tests

I added the following tests:
https://github.com/flutter/engine/pull/18065

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/foundation/_platform_web.dart [hunk 1]
```diff
@@ -2,15 +2,29 @@
...
+  final String navigatorPlatform = html.window.navigator.platform.toLowerCase();
+  if (navigatorPlatform.startsWith('mac')) {
+    return platform.TargetPlatform.macOS;
+  }
+  if (navigatorPlatform.contains('iphone') ||
+      navigatorPlatform.contains('ipad') ||
+      navigatorPlatform.contains('ipod')) {
+    return platform.TargetPlatform.iOS;
+  }
+  return platform.TargetPlatform.android;
```
Comment (nturgut) [inline packages/flutter/lib/src/foundation/_platform_web.dart:29]: are we considering desktop browsers android?

**Comment by jonahwilliams** [comment]
These are the only supported target platforms currently

File: packages/flutter/lib/src/foundation/_platform_web.dart (hunk 1 above)
Comment (ferhatb) [reply packages/flutter/lib/src/foundation/_platform_web.dart:29]: This PR makes sure scroll physics etc are correct on iOS/Mac. We will likely add to this, yes for now when running on fuchsia for example, it defaults to android physics.

**Review by nturgut — APPROVED** [review]

**Comment by AndroidNils** [comment]
Is there a ticket for the implementation of defaultTargetPlatform Windows/Linux (web)?

**Pull request merged**